src/tactic/core.lean

```diff
@@ -1145,6 +1145,47 @@ do e ← get_env,
 meta def is_in_mathlib (n : name) : tactic bool :=
 do ml ← get_mathlib_dir, e ← get_env, return $ e.is_prefix_of_file ml n
 
+meta def find_private_decl (n : name) (fr : option name) : tactic name :=
+do env ← get_env,
+   fn ← option_t.run (do
+         fr ← option_t.mk (return fr),
+         d ← monad_lift $ get_decl fr,
+         option_t.mk (return $ env.decl_olean d.to_name) ),
+   let p : string → bool :=
+     match fn with
+     | (some fn) := λ x, fn = x
+     | none := λ _, tt
+     end,
+   let xs := env.decl_filter_map (λ d,
+     do fn ← env.decl_olean d.to_name,
+        guard ((`_private).is_prefix_of d.to_name ∧ p fn ∧ d.to_name.update_prefix name.anonymous = n),
+        pure d.to_name),
+   match xs with
+   | [n] := pure n
+   | [] := fail "no such private found"
+   | _ := fail "many matches found"
+   end
+
+open lean.parser interactive
+
+/-- `import_private foo from bar` finds a private declaration in the same file as `bar` 
```

Comment:
```suggestion
/-- `import_private foo from bar` finds a private declaration `foo` in the same file as `bar` 
```

? I'm not totally clear how this interacts with namespacing. What if I declare `private def foo` in `my_namespace`, do I use `import_private my_namespace.foo`? Also, `from` taking another declaration in the same file seems strange. The most natural thing would be a module path, i.e. `import `private_nat_fn from data.nat.basic`, but that's probably harder to do.